cnxarchive/views.py

```diff
@@ -981,3 +1016,47 @@ def robots(request):
     resp.expires = html_date(exp_time)
     resp.body = robots_dot_txt.to_string()
     return resp
+
+@view_config(route_name='recent', request_method='GET',
+             accept="application/json",
+             renderer='templates/recent.rss') #renderer: cnxarchive.views:templates/recent.html?? also what should the permission actuall be??
+def recent(request):
+    # setting the query variables
+    query_vars = get_query_vars(request.query_string)
+    num_entries = 10
+    start_entry = 0
+    portal_type = "('Collection', 'Module')"
+    if "number" in query_vars.keys():
+        num_entries = query_vars["number"]
+    if "start" in query_vars.keys():
+        start_entry = query_vars["start"]
+    if "type" in query_vars.keys():
+        portal_type = query_vars["type"]
+    # search the database
+    settings = request.registry.settings
+    statement = """
+                SELECT name, revised, authors, abstract,
+                'http://cnx.org/contents/'||ident_hash( uuid, major_version, minor_version) AS link
+                FROM latest_modules
+                JOIN abstracts ON latest_modules.abstractid = abstracts.abstractid
+                WHERE portal_type in {}
+                ORDER BY revised DESC
+                LIMIT {} OFFSET {};
+                """.format(portal_type, num_entries, start_entry)
+    with psycopg2.connect(settings[config.CONNECTION_STRING]) as db_connection:
```

Comment:
Really important to do the param passing in execute, here. `number`, `start` and `type` are all passed in from the URL, so contain potentially hostile values. Needs all the `{}` replaced with `%s` then call:
`cursor.execute(statement, vars=portal_type, num_entries, start_entry))`